### ISSUE TITLE
RIDE should not allow editing read-only files

### ISSUE THREAD

**Issue by robotframework-ride** [robotframework/RIDE#192]
> *Originally submitted to [Google Code](http://code.google.com/p/robotframework-ride/issues/detail?id=157) by @yanne on 8 Jun 2009*

Read-only files could still be showed in the tree, but so that all the edit
controls would be disabled.

**Comment by robotframework-ride** [comment]
> *Originally submitted to [Google Code](http://code.google.com/p/robotframework-ride/issues/detail?id=157#c1) by bryan.oakley on 8 Jun 2009*


If you open a read-only file, you should be able to do a save-as to some other
location, and then be able to edit the files.

**Comment by robotframework-ride** [comment]
> *Originally submitted to [Google Code](http://code.google.com/p/robotframework-ride/issues/detail?id=157#c3) by @jussimalinen on 15 Apr 2011*


While we are waiting for better solutions, I would prefer at least some kind of a warning if I edit a read only file and try to save it. Now saving fails silently and I could lose(*) a lot of work if I dont notice that the star did not disappear from in front of the file.

(*) okay, so I did lose a lot of work because of this.

**Comment by robotframework-ride** [comment]
> *Originally submitted to [Google Code](http://code.google.com/p/robotframework-ride/issues/detail?id=157#c4) by @mkorpela on 9 Jun 2011*


Now that RIDE has run plugin (that requires that all the files have been saved) the severity of this (IMHO) _defect_ has raised.

Based on discussion in <a href="https://code.google.com/p/robotframework-ride/issues/detail?id=448" title="Undo (CTRL-Z) does not restore file state to not changed">Issue 448</a> I'll change the target and type.

**Comment by robotframework-ride** [comment]
> *Originally submitted to [Google Code](http://code.google.com/p/robotframework-ride/issues/detail?id=157#c5) by @mkorpela on 27 Feb 2012*


The backup mechanism gives more reasons to solve this issue sooner ( see <a class="closed_ref" href="https://code.google.com/p/robotframework-ride/issues/detail?id=939" title="No more possible to save TXT file from Ride when Backup fails"> issue 939 </a> ).

Thus pulling this to the next release.

**Comment by robotframework-ride** [comment]
> *Originally submitted to [Google Code](http://code.google.com/p/robotframework-ride/issues/detail?id=157#c7) by @mkorpela on 2 Mar 2012*


The number of modifying controls is very large, that is why I'm planning to implement the following functionality:
1) Show in tree and title that the current file is read only (for example "(READ ONLY)")
2) Fail all modification commands --&gt; publish Modification Failed message (this can be done in controller.execute)
3) Alert user about modification failure
4) Implement every modifying function as a command

**Comment by robotframework-ride** [comment]
> *Originally submitted to [Google Code](http://code.google.com/p/robotframework-ride/issues/detail?id=157#c8) by laurent.carbonnaux on 2 Mar 2012*


In my organization, all our automation files (TXT) are Read Only because they are stored in Perforce. They become writable when we check them out for modification. Hence this read-only topic is of very high importance for us.

Here aer my comments on your proposal :
1) adding (READY ONLY) in the tree/title of the test suite would make the tree less easy to read. If you want to add something, maybe it should be shorter (picture of a lock ?) and it should be at the beginning of the title (our titles are long, the end is not always visible)
2) could you not prevent the file to be modified at all ? For example on a read only file we should be able to modify any field (documentation, libraries etc.) and the table where the keywords are written should be locked as well. Maybe all those fiels should appear as grayed. If it is visually clear that the file is Read Only and the field are not editable, then having the ReadOnly informaton in the tree becomes less of a need.

I include a screenshot of the way RIDE looks like when we use it today.

Hope this helps to decide what to do.

**Comment by robotframework-ride** [comment]
> *Originally submitted to [Google Code](http://code.google.com/p/robotframework-ride/issues/detail?id=157#c9) by @mkorpela on 7 Mar 2012*


@&#8288;laurent:
1) I'll try to think of something else then.
2) Unfortunately solving this issue in that way requires a lot of work - compared to doing the bare minimum "alert when trying to modify" solution. And as this issue doesn't occur very often in my clients environments, I will first implement the bare minimum solution and add another issue about disabling all the controls in the read only case so that I can invest my time also on other issues in RIDE.

**Comment by robotframework-ride** [comment]
> *Originally submitted to [Google Code](http://code.google.com/p/robotframework-ride/issues/detail?id=157#c10) by laurent.carbonnaux on 7 Mar 2012*


@&#8288;mikko: ok, I understand your position. Thanks for your help !

**Comment by robotframework-ride** [comment]
> *Originally submitted to [Google Code](http://code.google.com/p/robotframework-ride/issues/detail?id=157#c11) by @mkorpela on 9 Mar 2012*


This is now done from my part:
<a href="https://github.com/robotframework/RIDE/commit/d68cb7b5ec983e7accfdf9bd54d55fcaf55501fd" rel="nofollow">https://github.com/robotframework/RIDE/commit/d68cb7b5ec983e7accfdf9bd54d55fcaf55501fd</a>

There are no indications in the tree about the read only status - this seems to be the case in other editors also.
Read only status is visible in the RIDE title text and also in the files header field.

Unfortunately this is currently a fail slow -solution that will indicate failure only after the user has tried to do a modifying action. Still the problems of loosing a lot of work by silent save failures or unexpected save working on read only files are now fixed.

**Comment by robotframework-ride** [comment]
> *Originally submitted to [Google Code](http://code.google.com/p/robotframework-ride/issues/detail?id=157#c12) by @mkorpela on 9 Mar 2012*


the new defect about controls being enabled is <a href="https://code.google.com/p/robotframework-ride/issues/detail?id=950" title="Editing controls should be disabled when file status is READ ONLY">issue 950</a> .
